### PR TITLE
Move code from examples tests

### PR DESCRIPTION
**Pull request by esteve**
Extracted code from the examples tests so that it can be reused more easily.

Connects to ros2/ros2#143

### CODE REVIEW

File: launch_testing/launch_testing/__init__.py [hunk 1]
```diff
@@ -72,3 +74,23 @@ def check(self):
...
             (output_lines, self.expected_lines)
+
+def create_handler(name, ld, output_file):
+    literal_file = output_file + '.txt'
+    regex_file = output_file + '.regex'
+    py_file = output_file + '.py'
+    if os.path.isfile(literal_file):
+        with open(literal_file, 'rb') as f:
+            expected_output = f.read().splitlines()
+        handler = InMemoryHandler(name, ld, expected_output, False)
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: Please pass keyword arguments with the keyword.

Same below.

File: launch_testing/launch_testing/__init__.py [hunk 2]
```diff
@@ -72,3 +74,23 @@ def check(self):
...
+    elif os.path.isfile(regex_file):
+        with open(regex_file, 'rb') as f:
+            expected_output = f.read().splitlines()
+        handler = InMemoryHandler(name, ld, expected_output, True)
+    elif os.path.isfile(py_file):
+        checker_module = SourceFileLoader(
+            'checker_module', py_file).load_module()
+        handler = checker_module.CheckerHandler(name, ld)
+    else:
+        assert False, 'Cannot find appropriate handler for %s' % output_file
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: Should the function return `None` instead and let the caller decide to assert?

File: launch_testing/launch_testing/__init__.py [hunk 3]
```diff
@@ -72,3 +74,23 @@ def check(self):
         assert self.matched, \
             'Example output (%r) does not match expected output (%r)' % \
             (output_lines, self.expected_lines)
+
+def create_handler(name, ld, output_file):
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: Either use a more descriptive argument name then `ld` or add a docblock describing it.

File: launch_testing/launch_testing/__init__.py (hunk 2 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Done. I also updated the code in the `test-services-examples` PR

File: launch_testing/launch_testing/__init__.py (hunk 3 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Done.

File: launch_testing/launch_testing/__init__.py [hunk 4]
```diff
@@ -72,3 +74,20 @@ def check(self):
...
+
+def create_handler(name, launch_descriptor, output_file):
+    literal_file = output_file + '.txt'
+    regex_file = output_file + '.regex'
+    py_file = output_file + '.py'
+    if os.path.isfile(literal_file):
+        with open(literal_file, 'rb') as f:
+            expected_output = f.read().splitlines()
+        return InMemoryHandler(name, launch_descriptor, expected_output, False)
+    elif os.path.isfile(regex_file):
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: `if` instead of `elif`.

Same below.

File: launch_testing/launch_testing/__init__.py [hunk 5]
```diff
@@ -72,3 +74,20 @@ def check(self):
         assert self.matched, \
             'Example output (%r) does not match expected output (%r)' % \
             (output_lines, self.expected_lines)
+
+def create_handler(name, launch_descriptor, output_file):
+    literal_file = output_file + '.txt'
+    regex_file = output_file + '.regex'
+    py_file = output_file + '.py'
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: These two lines can be moved further down to make their scope minimal.

**Comment by dirk-thomas** [comment]
This comment (https://github.com/ros2/launch/pull/15#discussion_r45539625) still applies.

File: launch_testing/launch_testing/__init__.py (hunk 1 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Done.

**Comment by esteve** [comment]
@dirk-thomas dude, just give me a minute, I was pushing the code :-)

File: launch_testing/launch_testing/__init__.py (hunk 5 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Done.

File: launch_testing/launch_testing/__init__.py (hunk 4 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Done.

File: launch_testing/launch_testing/__init__.py [hunk 6]
```diff
@@ -72,3 +74,21 @@ def check(self):
...
+    regex_file = output_file + '.regex'
+    if os.path.isfile(regex_file):
+        with open(regex_file, 'rb') as f:
+            expected_output = f.read().splitlines()
+        return InMemoryHandler(name, launch_descriptor, expected_output, regex_match=True)
+    py_file = output_file + '.py'
+    if os.path.isfile(py_file):
+        checker_module = SourceFileLoader(
+            'checker_module', py_file).load_module()
+        return checker_module.CheckerHandler(name, launch_descriptor)
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: Function lacks return statement.

**Comment by dirk-thomas** [comment]
Please lint the code.

File: launch_testing/launch_testing/__init__.py (hunk 6 above)
Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: All functions return `None` by default

**Comment by esteve** [comment]
Linters enabled and the code passes them.

**Comment by esteve** [comment]
Pushed as `test-services-examples`, CI jobs:

http://ci.ros2.org/job/ci_linux/626/
http://ci.ros2.org/job/ci_osx/515/
http://ci.ros2.org/job/ci_windows_connext_dynamic/14/
http://ci.ros2.org/job/ci_windows_connext_static/14/
http://ci.ros2.org/job/ci_windows_opensplice/697/

**Comment by esteve** [comment]
New tests pass on Linux and `launch_testing` passes the linters:

http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_add_two_ints_server__rmw_connext_cpp/
http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_add_two_ints_server__rmw_connext_dynamic_cpp/
http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_add_two_ints_server__rmw_opensplice_cpp/
http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_async_add_two_ints_server__rmw_connext_cpp/
http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_async_add_two_ints_server__rmw_connext_dynamic_cpp/
http://ci.ros2.org/job/ci_linux/626/testReport/%28root%29/test_add_two_ints_client_async_add_two_ints_server__rmw_opensplice_cpp/

**Comment by dirk-thomas** [comment]
lgtm.

File: launch_testing/launch_testing/__init__.py [hunk 7]
```diff
@@ -72,3 +76,22 @@ def check(self):
         assert self.matched, \
             'Example output (%r) does not match expected output (%r)' % \
             (output_lines, self.expected_lines)
+
+
+def create_handler(name, launch_descriptor, output_file):
+    literal_file = output_file + '.txt'
+    py_file = output_file + '.py'
```
Comment (dirk-thomas) [inline launch_testing/launch_testing/__init__.py]: Duplicate line removed in 3b4ea2ea31d9748cd7a127b4229c723b962ba735.

Comment (esteve) [inline launch_testing/launch_testing/__init__.py]: Thanks!

**Pull request merged**